control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -0,0 +1,335 @@
+				return ctrl.Result{}, err
+			}
+		}
+		// Delete the existing secret if the name changed. This needs to come before updating the status if we do generate a new token.
+		if nameChanged {
+			err := r.Client.Delete(ctx, existingSecret)
+			if err != nil {
+				_ = r.updateStatusError(ctx, peeringAcceptor, err)
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Store the state in the status.
+		err := r.updateStatus(ctx, peeringAcceptor, resp)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// shouldGenerateToken returns whether a token should be generated, and whether the name of the secret has changed. It
+// compares the spec secret's name/key/backend and contents to the status secret's name/key/backend and contents. The
+// contents are compared by taking a SHA256 sum of the secret.
+func shouldGenerateToken(peeringAcceptor *consulv1alpha1.PeeringAcceptor, existingSecret *corev1.Secret) (shouldGenerate bool, nameChanged bool, err error) {
+	if peeringAcceptor.Status.Secret == nil {
+		return false, false, errors.New("shouldGenerateToken was called with an empty fields in the existing status")
+	}
+	// Compare the existing name, key, and backend.
+	if peeringAcceptor.Status.Secret.Name != peeringAcceptor.Spec.Peer.Secret.Name {
+		return true, true, nil
+	}
+	if peeringAcceptor.Status.Secret.Key != peeringAcceptor.Spec.Peer.Secret.Key {
+		return true, false, nil
+	}
+
+	// TODO(peering): remove this when validation webhook exists.
+	if peeringAcceptor.Status.Secret.Backend != peeringAcceptor.Spec.Peer.Secret.Backend {
+		return false, false, errors.New("PeeringAcceptor backend cannot be changed")
+	}
+	// Compare the existing secret hash.
+	// Get the secret specified by the status, make sure it matches the status' secret.latestHash.
+	if existingSecret != nil {
+		existingSecretHashBytes := sha256.Sum256(existingSecret.Data[peeringAcceptor.Status.Secret.Key])
+		existingSecretHash := hex.EncodeToString(existingSecretHashBytes[:])
+		if existingSecretHash != peeringAcceptor.Status.Secret.LatestHash {
+			return true, false, nil
+		}
+
+	}
+	return false, false, nil
+}
+
+// updateStatusError updates the peeringAcceptor's secret in the status.
+func (r *PeeringAcceptorController) updateStatus(ctx context.Context, peeringAcceptor *consulv1alpha1.PeeringAcceptor, resp *api.PeeringGenerateTokenResponse) error {
+	peeringAcceptor.Status.Secret = &consulv1alpha1.SecretStatus{
+		Name:    peeringAcceptor.Spec.Peer.Secret.Name,
+		Key:     peeringAcceptor.Spec.Peer.Secret.Key,
+		Backend: peeringAcceptor.Spec.Peer.Secret.Backend,
+	}
+
+	peeringTokenHash := sha256.Sum256([]byte(resp.PeeringToken))
+	peeringAcceptor.Status.Secret.LatestHash = hex.EncodeToString(peeringTokenHash[:])
+
+	peeringAcceptor.Status.LastReconcileTime = &metav1.Time{Time: time.Now()}
+	peeringAcceptor.Status.ReconcileError = &consulv1alpha1.ReconcileErrorStatus{
+		Error:   pointerToBool2(false),
```

Comment:
should we use `pointerToBool` (I'm assuming we already have this function in this package - that's why it's named like this?)

Reply:
Omg yeah whoops I had done that so I could run tests using just the two file names (controller and controller_test) without needing to include the file that has pointerToBool! Thanks for catching!!